src/Workspaces/Core/Portable/Execution/AbstractReferenceSerializationService.cs

```diff
@@ -531,7 +531,10 @@ private void WriteTo(AnalyzerReference reference, ObjectWriter writer, bool chec
                 // snapshot version for analyzer (since it is based on shadow copy)
                 // we can't send over bits and load analyer from memory (image) due to CLR not being able
                 // to find satellite dlls for analyzers.
-                writer.WriteString(assemblyPath);
+                //
+                // make sure we always normalize assemblyPath to lower case since it comes from Assembly.Location
+                // unlike FullPath which comes from string
+                writer.WriteString(assemblyPath.ToLower());
```

Comment:
Culture can change within the same process. So you write this, culture changes, and now this serialization is wrong. You never want to persist using a format that can change.